### PR TITLE
[WIP] Version selector for documentation.

### PR DESCRIPTION
**Pull request by Carreau**
That's still highly in progress, for now poking around to see how it could look like. 

This basically setup a marker for the version-switcher in the sidebar and search/replace it script.
I try to be smart enough and if the page does not exists in another version of the docs, it redirect to `index.html`. The main point of research/replace being that you can update "old" docs without rebuilding. 

We will need at some point to move the documentation to a subfolder like /stable/, maybe set a meta-redirect, but this is extremely frowned upon and not good for SEO.

### CODE REVIEW

**Comment by scopatz** [comment]
This would be awesome to have! Just let us know when this is no longer WIP or if you need us to test it out locally.

**Comment by Carreau** [comment]
Meh... I'm unsure it should works but doctr does trigger on PR...

**Comment by Carreau** [comment]
I'll see if I can patch doctr to actually have a dry-run and not push on PR instead of Just bailing out.

File: .travis.yml [hunk 1]
```diff
@@ -46,11 +46,16 @@ install:
 
 script:
   - if [[ $BUILD_DOCS = true ]]; then
+      echo "buildign docs"
```
Comment (Carreau) [inline .travis.yml]: Yeah.. just debug... trying stuff. I'll rbase clean later once I'm happy. I might do a few change to doctr first.

**Comment by Carreau** [comment]
I think I know have an idea of all the changes that are needed in doctr to deploy that. It will take me a couple of days to finish implementing that and maybe get a release of doctr.

**Comment by scopatz** [comment]
I am really looking forward to it!

**Comment by scopatz** [comment]
Hi @Carreau - I am pretty sure more releases of doctr have come out :) Is this more or less ready to go?

**Comment by Carreau** [comment]
I'll hack on that on the week of March 6, we're doing a docathon in BIDS, eScience Institute and NY.

**Comment by scopatz** [comment]
Sounds like a plan and like fun! Thanks @Carreau

**Comment by Carreau** [comment]
Note to myself: https://pypi.python.org/pypi/sphinxcontrib-versioning

**Comment by gforsyth** [comment]
Hey @Carreau -- should we leave this PR open?  Do you think you'll have time at some point to shepherd this through?

**Comment by Carreau** [comment]
let's close it for now, I can come back to it later and reopen.

Thanks !

**Pull request closed without merging**